### PR TITLE
Add 'ignore edited messages' condition to automod v2

### PR DESCRIPTION
**Pull request by Alecat**
I mocked up this change for a feature that would be useful on my server. I have not yet been able to run/test this code but I saw that my suggestion got pinned on the server and if this helps to get it implemented faster I hope I can save the dev team's time.

Hopefully it's on the right track and doesn't need too much change in order to be implemented/useful.

### CODE REVIEW

File: automod/conditions.go [hunk 1]
```diff
@@ -1,9 +1,10 @@
 package automod
 
 import (
+	"time"
+
```
Comment (Alecat) [inline automod/conditions.go]: Oops, remove import

File: automod/conditions.go [hunk 2]
```diff
@@ -475,3 +476,69 @@ func (bc *BotCondition) IsMet(data *TriggeredRuleData, settings interface{}) (bo
...
+func (mc *MessageEditedCondition) Name() string {
+	if mc.NewMessage {
+		return "New message"
+	}
+
+	return "Edited message"
+}
+
+func (mc *MessageEditedCondition) Description() string {
+	if mc.Below {
```
Comment (Alecat) [inline automod/conditions.go]: NewMessage

**Comment by jonas747** [comment]
Seems like you're in the right direction atleast, ill test it before i merge it
also you need to add it to this listing for it to show up anywhere: https://github.com/jonas747/yagpdb/blob/master/automod/rulepart.go#L46

File: automod/conditions.go [hunk 3]
```diff
@@ -475,3 +476,65 @@ func (bc *BotCondition) IsMet(data *TriggeredRuleData, settings interface{}) (bo
 func (bc *BotCondition) MergeDuplicates(data []interface{}) interface{} {
 	return data[0] // no point in having duplicates of this
 }
+
+/////////////////////////////////////////////////////////////////
+
+type MessageEditedData struct {
```
Comment (jonas747) [inline automod/conditions.go]: This condition needs no data, so remove this and have DataType return nil

File: automod/rulepart.go [hunk 4]
```diff
@@ -56,6 +56,7 @@ var RulePartMap = map[int]RulePart{
 	210: &BotCondition{Ignore: false},
 	211: &ChannelCategoriesCondition{Blacklist: true},
 	212: &ChannelCategoriesCondition{Blacklist: false},
+	213: &MessageEditedCondition{NewMessage: true},
```
Comment (jonas747) [inline automod/rulepart.go]: Add one for `NewMessage: false` aswell

File: automod/conditions.go [hunk 5]
```diff
@@ -475,3 +476,65 @@ func (bc *BotCondition) IsMet(data *TriggeredRuleData, settings interface{}) (bo
...
+	}
+	if data.Message.EditedTimestamp == "" {
+		// new post
+		if mc.NewMessage {
+			return true, nil
+		}
+		return false, nil
+	}
+
+	// account is older than threshold
```
Comment (jonas747) [inline automod/conditions.go]: forgot about this comment?

**Comment by jonas747** [comment]
Looks good, in the future tough you should run go fmt on it, but for now ill merge this and run that myself.

**Comment by Alecat** [comment]
Cheers! I actually did have my editor auto-format on save, but I wasn't sure about pushing up the reordered imports so I only pushed my lines :)

Thanks for the code review and for maintaining YAGPDB!

**Comment by jonas747** [comment]
Ah i see, its aight.

**Pull request merged**